exporters/etw/include/opentelemetry/exporters/etw/etw_tracer.h

```diff
@@ -235,9 +236,17 @@ class Tracer : public opentelemetry::trace::Tracer,
                        const opentelemetry::trace::Span *parentSpan = nullptr,
                        const opentelemetry::trace::EndSpanOptions & = {})
   {
-    const auto &cfg = GetConfiguration(tracerProvider_);
+    const auto &cfg          = GetConfiguration(tracerProvider_);
+    const auto &tail_sampler = GetTailSampler(tracerProvider_);
     const opentelemetry::trace::Span &spanBase =
         reinterpret_cast<const opentelemetry::trace::Span &>(span);
+
+    //  Sample span based on the decision of tail based sampler
+    auto sampling_result = const_cast<TailSampler *>(&tail_sampler)->ShouldSample(spanBase);
```

Comment:
Instead of making a virtual call to AlwaysOn sampler by default, perhaps check if there is a non-default sampler set, then make the virtual call? In this way, it doesn't introduce extra cost (except the null check) when tail based sampler is not used.

Reply:
Good comment. Actually we do have virtual method calls for head-sampler, id-generator both in ETW exporter, and the otel-sdk. There is no reported performance reported with them. Seemingly modern compilers can optimize these repeated calls by caching the address, and avoid virtual indirection. Perhaps we can look back to this if there are reported overhead internally and/or by community.